### PR TITLE
gui: Fix help dialogue

### PR DESCRIPTION
**Pull request by nishinji**
before
image

after
image
image

### CODE REVIEW

**Comment by Zangetsu38** [comment]
Code if is wrong

**Comment by nishinji** [comment]
you mean
`!exist || empty`?

**Comment by Zangetsu38** [comment]
> you mean `!exist || empty`?

i mean your both if code is is wrong

File: vita3k/gui/src/live_area.cpp [hunk 1]
```diff
@@ -1057,10 +1057,17 @@ void draw_live_area_screen(GuiState &gui, EmuEnvState &emuenv) {
             ImGui::Spacing();
             ImGui::SetCursorPosX(ImGui::GetWindowWidth() / 2.f - (ImGui::CalcTextSize("Using configuration set for keyboard in control setting").x / 2.f));
             ImGui::TextColored(GUI_COLOR_TEXT, "Using configuration set for keyboard in control setting");
-            if (gui.modules.empty()) {
+            const auto FW_PATH{ fs::path(emuenv.pref_path) / "vs0" };
+            const auto FW_FONT_PATH{ fs::path(emuenv.pref_path) / "sa0" };
+            if (fs::exists(FW_PATH) && fs::is_empty(FW_PATH)) {
```
Comment (Zangetsu38) [inline vita3k/gui/src/live_area.cpp:1062]:
```suggestion
    if (gui.modules.empty()) {
```
keep it

File: vita3k/gui/src/live_area.cpp [hunk 2]
```diff
@@ -1057,10 +1057,17 @@ void draw_live_area_screen(GuiState &gui, EmuEnvState &emuenv) {
...
                 ImGui::Spacing();
-                ImGui::SetCursorPosX(ImGui::GetWindowWidth() / 2.f - (ImGui::CalcTextSize("Firmware not detected. Installing it is recommended for font text in Live Area").x / 2.f));
-                ImGui::TextColored(GUI_COLOR_TEXT, "Firmware not detected. Installing it is recommended for font text in Live Area");
+                ImGui::SetCursorPosX(ImGui::GetWindowWidth() / 2.f - (ImGui::CalcTextSize("Firmware not detected. It is required to launch applications").x / 2.f));
+                ImGui::TextColored(GUI_COLOR_TEXT, "Firmware not detected. It is required to launch applications");
+            }
+            if (fs::exists(FW_FONT_PATH) && fs::is_empty(FW_FONT_PATH)) {
```
Comment (Zangetsu38) [inline vita3k/gui/src/live_area.cpp:1067]:
```suggestion
            if (!gui.vita_font) {
```
check for it, i have forget name lol

File: vita3k/gui/src/live_area.cpp [hunk 3]
```diff
@@ -1057,10 +1057,17 @@ void draw_live_area_screen(GuiState &gui, EmuEnvState &emuenv) {
...
             ImGui::TextColored(GUI_COLOR_TEXT, "Using configuration set for keyboard in control setting");
-            if (gui.modules.empty()) {
+            const auto FW_PATH{ fs::path(emuenv.pref_path) / "vs0" };
+            const auto FW_FONT_PATH{ fs::path(emuenv.pref_path) / "sa0" };
+            if (fs::exists(FW_PATH) && fs::is_empty(FW_PATH)) {
                 ImGui::Spacing();
-                ImGui::SetCursorPosX(ImGui::GetWindowWidth() / 2.f - (ImGui::CalcTextSize("Firmware not detected. Installing it is recommended for font text in Live Area").x / 2.f));
-                ImGui::TextColored(GUI_COLOR_TEXT, "Firmware not detected. Installing it is recommended for font text in Live Area");
+                ImGui::SetCursorPosX(ImGui::GetWindowWidth() / 2.f - (ImGui::CalcTextSize("Firmware not detected. It is required to launch applications").x / 2.f));
+                ImGui::TextColored(GUI_COLOR_TEXT, "Firmware not detected. It is required to launch applications");
```
Comment (Zangetsu38) [inline vita3k/gui/src/live_area.cpp:1065]: little wrong, fw is not requiered, very depending game, is recomanded and obloigatory for lot game, but not all

File: vita3k/gui/src/live_area.cpp (hunk 1 above)
Comment (nishinji) [reply vita3k/gui/src/live_area.cpp:1062]: The text is displayed without loading the module even if the firmware is installed

Comment (Croden1999) [reply vita3k/gui/src/live_area.cpp:1062]: > The text is displayed without loading the module even if the firmware is installed

only module, almost files does not unused and work in vita3k.

Comment (nishinji) [reply vita3k/gui/src/live_area.cpp:1062]: ??
My point is that the module is not always loaded while the emulator is running.

Comment (Croden1999) [reply vita3k/gui/src/live_area.cpp:1062]: > ?? My point is that the module is not always loaded while the emulator is running.

but load modules in some games are needed.

Comment (Zangetsu38) [reply vita3k/gui/src/live_area.cpp:1062]: I repeat your carrément code is wrong, your if condition is wrong
And so no need check folder

**Review by Zangetsu38 — APPROVED** [review]

**Pull request merged**